### PR TITLE
Introduces categorized show and edit for system variables

### PR DESCRIPTION
**Pull request by ghatighorias**
1. Separates the system_variables to system and program and uses two different tables to list them in the show page
2. introduces two separate edit link for editing system and program related system variables separately
3. updates the router so edit can get an id to select editing between system or program system_variables
4. updates the current tests and introduces new ones

### CODE REVIEW

File: web/controllers/setting_controller.ex [hunk 1]
```diff
@@ -8,16 +8,39 @@ defmodule CoursePlanner.SettingController do
...
+        |> put_status(404)
+        |> render(CoursePlanner.ErrorView, "404.html")
+      _   ->
+        changeset =
+          filtered_system_variables
+          |> Enum.map(&SystemVariable.changeset/1)
+          |> Settings.wrap()
+
+        render(conn, "edit.html", changeset: changeset)
+    end
```
Comment (rhnonose) [inline web/controllers/setting_controller.ex]: Can you trim this function a bit? It's too long.
Looks like you can "merge" both `case` statements and use a separate function to extract the `"setting_type"` or simply split it into two functions.

Comment (ghatighorias) [inline web/controllers/setting_controller.ex]: I moved parts of the functionality out, check if you see it good enough

**Comment by ggpasqualino** [comment]
@ghatighorias please check the conflicts

File: test/controllers/setting_controller_test.exs [hunk 2]
```diff
@@ -116,20 +127,23 @@ defmodule CoursePlanner.SettingControllerTest do
     end
   end
 
+  @tag user_role: :volunteer
```
Comment (ggpasqualino) [inline test/controllers/setting_controller_test.exs]: should this be `@module_tag` as well?

File: web/models/setting/settings.ex [hunk 3]
```diff
@@ -37,6 +37,22 @@ defmodule CoursePlanner.Settings do
...
+  def filter_system_variables(system_variables, setting_type) do
+      case setting_type do
+        "system_settings"  -> {:ok, filter_non_program_systemvariables(system_variables)}
+        "program_settings" -> {:ok, filter_program_systemvariables(system_variables)}
+        _                  -> {:error, nil}
+      end
+  end
+
+  def filter_non_program_systemvariables(settings) do
+    Enum.filter(settings, &(not String.starts_with?(&1.key, "PROGRAM")))
```
Comment (ggpasqualino) [inline web/models/setting/settings.ex]: please avoid negative conditions, what do you think of the following?
```elixir
Enum.reject(settings, &(String.starts_with?(&1.key, "PROGRAM")))
```

File: web/router.ex [hunk 4]
```diff
@@ -74,7 +74,9 @@ defmodule CoursePlanner.Router do
       put "/update_fill", AttendanceController, :update_fill, as: :update_fill
     end
 
-    resources "/settings", SettingController, only: [:show, :edit, :update], singleton: true
+    resources "/settings", SettingController, only: [:show, :update], singleton: true do
```
Comment (ggpasqualino) [inline web/router.ex]: if you don't change the router you can achieve a similar result using the following links
```elixir
setting_path(@conn, :edit, setting_type: "system") # /settings?setting_type=system
```
or
```elixir
setting_path(@conn, :edit, setting_type: "program") # /settings?setting_type=program
```
the controller can stay the same for this urls

the advantage would be having a standard path helper function

File: test/controllers/setting_controller_test.exs (hunk 2 above)
Comment (ghatighorias) [inline test/controllers/setting_controller_test.exs]: done

File: web/models/setting/settings.ex (hunk 3 above)
Comment (ghatighorias) [inline web/models/setting/settings.ex]: done

File: web/router.ex (hunk 4 above)
Comment (ghatighorias) [inline web/router.ex]: nice trick, i'll take a note on this one

Comment (ghatighorias) [inline web/router.ex]: done

**Pull request merged**